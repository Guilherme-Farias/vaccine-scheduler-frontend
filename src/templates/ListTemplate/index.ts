export { default } from './ListTemplate';
export * from './ListTemplate';
