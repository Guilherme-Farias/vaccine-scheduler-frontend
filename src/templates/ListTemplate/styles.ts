import styled, { css } from 'styled-components/macro';

export const Container = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings['2xl']};
  `}
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes['3xl']};
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xl};
  `}
`;
