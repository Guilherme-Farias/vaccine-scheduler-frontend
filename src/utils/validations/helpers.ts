import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
