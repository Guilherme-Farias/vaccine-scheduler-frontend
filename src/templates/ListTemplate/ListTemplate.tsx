import React from 'react';

import MaxWidthWrapper from 'components/MaxWidthWrapper';
import Spinner from 'components/Spinner';

import * as S from './styles';

export type ListTemplateProps = {
  title: string;
  description: string;
  children: React.ReactNode;
  isLoading?: boolean;
};

const ListTemplate: React.FC<ListTemplateProps> = ({
  title,
  description,
  children,
  isLoading,
}: ListTemplateProps) => (
  <MaxWidthWrapper>
    <S.Container>
      <div
        style={{
          display: 'flex',
          alignItems: 'center',
          gap: '10px',
        }}
      >
        <S.Title>{title}</S.Title>
        {isLoading && <Spinner />}
      </div>
      <S.Description>{description}</S.Description>
    </S.Container>
    {children}
  </MaxWidthWrapper>
);

export default ListTemplate;
