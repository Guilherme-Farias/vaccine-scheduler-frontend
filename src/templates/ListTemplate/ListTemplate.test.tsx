import { render, screen } from 'utils/tests/helpers';

import ListTemplate from '.';

describe('<ListTemplate />', () => {
  it('should render the ListTemplate', () => {
    const { container } = render(
      <ListTemplate title="Title" description="Description">
        <p>Children</p>
      </ListTemplate>,
    );
    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument();
    expect(screen.getByText(/description/i)).toBeInTheDocument();
    expect(screen.getByText(/children/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});

export default 1;
