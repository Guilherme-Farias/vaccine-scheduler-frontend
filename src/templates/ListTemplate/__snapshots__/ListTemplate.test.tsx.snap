// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListTemplate /> should render the ListTemplate 1`] = `
.c0 {
  width: 100%;
  max-width: 81.25rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

.c1 {
  margin-bottom: 1.75rem;
}

.c2 {
  font-size: 1.875rem;
}

.c3 {
  font-size: 1.25rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      style="display: flex; align-items: center; gap: 10px;"
    >
      <h2
        class="c2"
      >
        Title
      </h2>
    </div>
    <p
      class="c3"
    >
      Description
    </p>
  </div>
  <p>
    Children
  </p>
</div>
`;
